examples/send_file.rs

```diff
@@ -106,38 +58,38 @@ fn simple_file_send(f: &str) -> ResponseFuture {
     // this is limited to serving small files, but it is somewhat
     // simpler with a little less overhead.
     //
-    // On channel errors, we panic with the expect method. The thread
-    // ends at that point in any case.
+    // Wrap in tokio_threadpool::blocking to tell the ThreadPool
+    // that the call to read_file_blocking will block.
     let filename = f.to_string(); // we need to copy for lifetime issues
-    let (tx, rx) = oneshot::channel();
-    thread::spawn(move || {
-        let mut file = match File::open(filename) {
-            Ok(f) => f,
-            Err(_) => {
-                tx.send(Response::builder()
-                        .status(StatusCode::NOT_FOUND)
-                        .body(NOTFOUND.into())
-                        .unwrap())
-                    .expect("Send error on open");
-                return;
-            },
-        };
-        let mut buf: Vec<u8> = Vec::new();
-        match copy(&mut file, &mut buf) {
-            Ok(_) => {
-                let res = Response::new(buf.into());
-                tx.send(res).expect("Send error on successful file read");
-            },
-            Err(_) => {
-                tx.send(Response::builder()
-                        .status(StatusCode::INTERNAL_SERVER_ERROR)
-                        .body(Body::empty())
-                        .unwrap())
-                    .expect("Send error on error reading file");
-            },
-        };
-    });
-
-    Box::new(rx.map_err(|e| io::Error::new(io::ErrorKind::Other, e)))
+    Box::new(
+        future::poll_fn(move || {
+            blocking(|| read_file_blocking(&filename))
```

Comment:
Should this instead show just using `tokio-fs` instead? I slightly worry people will miss this line and think they should just do blocking operations anywhere... Or do you think there is value in seeing the more general `blocking` tool?

Reply:
Fair point.  I updated the example to use tokio_fs and tokio_io to have the same behavior, which is probably a better example.